src/test/integration/IntegrationBase.t.sol

```diff
@@ -351,6 +347,234 @@ abstract contract IntegrationBase is IntegrationDeployer {
+        Allocation[] memory prevAllocs = _getPrevAllocations(op, operatorSet, slashingParams.strategies);
+        Magnitudes[] memory curMagnitudes = _getMagnitudes(op, slashingParams.strategies);
+        Magnitudes[] memory prevMagnitudes = _getPrevMagnitudes(op, slashingParams.strategies);
+
+        (, uint32 delay) = allocationManager.getAllocationDelay(slashingParams.operator);
+
+        for (uint i = 0; i < slashingParams.strategies.length; i++) {
+            Allocation memory curAlloc = curAllocs[i];
+            Allocation memory prevAlloc = prevAllocs[i]; 
+
+            uint64 slashedMagnitude = uint64(uint256(prevAlloc.currentMagnitude).mulWadRoundUp(slashingParams.wadsToSlash[i]));
+
+            assertEq(
+                curAlloc.currentMagnitude, 
+                prevAlloc.currentMagnitude - slashedMagnitude, 
+                string.concat(err, " (currentMagnitude)")
+            );
+
+            if (completed) {
+                assertEq(curAlloc.pendingDiff, 0, string.concat(err, " (pendingDiff)"));
+                assertEq(curAlloc.effectBlock, 0, string.concat(err, " (effectBlock)"));
+            } else {
+                assertEq(
+                    curAlloc.currentMagnitude, 
+                    prevAlloc.currentMagnitude - slashedMagnitude, 
+                    string.concat(err, " (currentMagnitude)")
+                );
+
+                if (prevAlloc.pendingDiff < 0) {
+                    uint64 slashedPending =
+                        uint64(uint256(uint128(-prevAlloc.pendingDiff)).mulWadRoundUp(slashingParams.wadsToSlash[i]));
+                    
+                    assertEq(
+                        curAlloc.pendingDiff, 
+                        prevAlloc.pendingDiff + int128(uint128(slashedPending)), 
+                        string.concat(err, " (pendingDiff)")
+                    );
+                }
+
+                assertEq(
+                    curAlloc.effectBlock, 
+                    block.number + delay, 
+                    string.concat(err, " (effectBlock)")
+                );
+            }
+
+            Magnitudes memory curMagnitude = curMagnitudes[i];
+            Magnitudes memory prevMagnitude = prevMagnitudes[i];
+
+            assertEq(
+                curMagnitude.encumbered, 
+                prevMagnitude.encumbered - slashedMagnitude, 
+                string.concat(err, " (encumbered magnitude)")
+            );
+
+            assertEq(
+                curMagnitude.allocatable, 
+                prevMagnitude.allocatable, 
+                string.concat(err, " (allocatable magnitude)")
+            );
+
+            assertEq(
+                curMagnitude.max, 
+                prevMagnitude.max - slashedMagnitude, 
+                string.concat(err, " (max magnitude)")
+            );
+        }
+    }
+
+    function assert_HasUnderlyingTokenBalances_AfterSlash(
```

Comment:
For non native ETH Strategies, we should also have an assertion on the token balance of the Strategy being burned/updated after the `slashOperator` call and it should be sent to the burn address.
Perhaps it makes sense to be in this check or a separate assert function

Reply:
will include in follow up pr